src/Consolonia.Core/Text/Fonts/EmbeddedConsoleFontCollection.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            if (TryGetGlyphTypeface(familyName, fontStyle, fontWeight, fontStretch, out IGlyphTypeface glyphTypeface))
+                if (glyphTypeface.TryGetGlyph((uint)codepoint, out _))
+                {
+                    typeface = new Typeface(familyName, fontStyle, fontWeight, fontStretch);
+                    return true;
+                }
+
+            typeface = default;
+            return false;
+        }
+        // Resharper enable AssignNullToNotNullAttribute
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        /// <summary>
+        ///     Load a typeface from an embedded resource by name
+        /// </summary>
+        /// <param name="familyName"></param>
+        /// <param name="resources"></param>
+        /// <returns></returns>
+        protected IGlyphTypeface LoadEmbeddedFontFamily(string familyName, params Uri[] resources)
+        {
+            var familyTypespace = new AsciiArtFamilyTypeface(familyName);
+
+            foreach (Uri resource in resources)
+            {
+                if (!_typefaceByUri.TryGetValue(resource, out IGlyphTypeface typeface))
+                    typeface = LoadEmbeddedFont(resource);
+                familyTypespace.AddTypeface(typeface);
+            }
+
+            _typefaceByName[familyName] = familyTypespace;
+            return familyTypespace;
+        }
+
+        /// <summary>
+        ///     Load an embedded font from a given asset URI
+        /// </summary>
+        /// <param name="uri">avares: uri to the font</param>
+        /// <returns></returns>
+        protected IGlyphTypeface LoadEmbeddedFont(Uri uri)
+        {
+            string resourceName = Path.GetFileName(uri.AbsolutePath);
+            string fontName = Path.GetFileNameWithoutExtension(uri.AbsolutePath);
+            using Stream stream = AssetLoader.Open(uri);
+            IGlyphTypeface typeface =
+                AsciiArtTypefaceLoader.Load(stream, Path.GetFileNameWithoutExtension(resourceName));
+            if (typeface == null)
+                throw new InvalidOperationException($"Failed to load font variation: {resourceName}");
+            _typefaceByName[fontName] = typeface;
+            _typefaceByUri[uri] = typeface;
+            return typeface;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    // TODO: dispose managed state (managed objects)
+                    foreach (IGlyphTypeface typeface in _typefaceByUri.Values.Distinct()) typeface.Dispose();
+                    _typefaceByUri.Clear();
+                    _typefaceByName.Clear();
+                    _fontFamilies.Clear();
+                }
+
+                // TODO: free unmanaged resources (unmanaged objects) and override finalizer
```

Comment:
These todos and commented code does not suggest to do anything from our side, it seems no use of it.

Reply:
these are generated by visual studio, but I can clean them up